pipeline-model-definition/src/test/java/org/jenkinsci/plugins/pipeline/modeldefinition/parser/ModelParserTest.java

```diff
@@ -63,6 +63,35 @@ public void labelWithOptionsBecomesNode() throws Exception {
         assertEquals(nodeRoot, newRoot);
     }
 
+    @Issue("JENKINS-43016")
+    @Test
+    public void labelWithEmptyStringBecomesAny() throws Exception {
+        ModelASTPipelineDef origRoot = Converter.urlToPipelineDef(getClass().getResource("/agentLabelEmptyString.groovy"));
+
+        assertNotNull(origRoot);
+
+        JSONObject origJson = origRoot.toJSON();
+        assertNotNull(origJson);
+
+        JSONParser jp = new JSONParser(Converter.jsonTreeFromJSONObject(origJson));
+        ModelASTPipelineDef newRoot = jp.parse();
+
+        assertEquals(getJSONErrorReport(jp, "agentLabelEmptyString"), 0, jp.getErrorCollector().getErrorCount());
+        assertNotNull("Pipeline null for agentLabelEmptyString", newRoot);
+
+        JSONObject anyJson = JSONObject.fromObject(fileContentsFromResources("json/agentAny.json"));
+
+        JSONParser anyParser = new JSONParser(Converter.jsonTreeFromJSONObject(anyJson));
+        ModelASTPipelineDef anyRoot = anyParser.parse();
+
+        assertEquals(getJSONErrorReport(anyParser, "agentAny"),
+                0, anyParser.getErrorCollector().getErrorCount());
```

Comment:
I see you checking that there are no errors, and that might be the actual thing to test. But shouldn't you check that it's actually `agent any` that it has been converted to and just not something that doesn't have errors? I mean `agent label ' '` might not have errors, but actually not empty?